sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -60,7 +66,7 @@ def _send_request(self, relative_url, accepted_status_codes, function, data=None
     def _send_request_helper(self, url, accepted_status_codes, function, data, retry_count):
         while True:
             try:
-                if self._endpoint.auth == constants.NO_AUTH:
+                if self._endpoint.auth.login_service == constants.NO_AUTH:
```

Comment:
So I think this means the base authenticator class instance would need to be none correct? So then I am thinking this would require an if check in the for loop that builds the auth_instances map mapping the class path string to the instances of the class and explicitly checking if it is the base class, then just map it to None. Does that sound right?